src/rgw/rgw_s3_filter.cc

```diff
@@ -45,18 +54,28 @@ bool rgw_s3_key_filter::decode_xml(XMLObj* obj) {
   auto suffix_not_set = true;
   auto regex_not_set = true;
   std::string name;
+  std::string type; 
```

Comment:
can you please rename this to "exclude"?
also, can this be boolean instead of string? do we have proper xml encode/decode options for booleans?